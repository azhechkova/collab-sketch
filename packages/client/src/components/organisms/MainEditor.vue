<script setup lang="ts">
import type { Socket } from 'socket.io-client'
import TheEditor from './TheEditor.vue'
import TheLegend from './TheLegend.vue'
import { inject } from 'vue'

const socket = inject('socket') as Socket
</script>

<template>
  <div class="flex gap-2">
    <TheEditor class="flex-shrink-0" :socket="socket" />
    <TheLegend :socket="socket" />
  </div>
</template>
