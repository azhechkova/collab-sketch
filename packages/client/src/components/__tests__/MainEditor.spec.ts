import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'

import MainEditor from '../atoms/MainEditor.vue'

describe('Main Editor', () => {
  it('renders properly', () => {
    const wrapper = mount(MainEditor)
    expect(wrapper.text()).toMatchSnapshot()
  })

  it('renders the canvas element', async () => {
    const wrapper = mount(MainEditor)

    const canvas = wrapper.find('canvas')
    expect(canvas.exists()).toBe(true)
  })

  it('stops drawing when the mouse button is released', async () => {
    const wrapper = mount(MainEditor)
    const canvas = wrapper.find('canvas')

    await canvas.trigger('mousedown', { offsetX: 10, offsetY: 10 })
    await canvas.trigger('mousemove', { offsetX: 20, offsetY: 20 })
    await canvas.trigger('mouseup')

    expect(wrapper.vm.isDrawing).toBe(false)
  })
})
