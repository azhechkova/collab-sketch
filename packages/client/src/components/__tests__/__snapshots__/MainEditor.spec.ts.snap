// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main Editor > renders properly 1`] = `""`;
